featuretools/tests/entityset_tests/test_es.py

```diff
@@ -1336,6 +1336,21 @@ def test_secondary_time_index(es):
             'second_ti': ['comments', 'second_ti']})
 
 
+def test_set_secondary_time_index_dataframe_not_present(es):
+    error = 'DataFrame not_present does not exist in ecommerce'
+    with pytest.raises(KeyError, match=error):
+        es.set_secondary_time_index('not_present', {'dates': ['id']})
+
+
+def test_set_secondary_time_index_on_external_dataframe(es):
+    external_df = pd.DataFrame({'id': [1, 2, 3], 'dates': ['2020-08-01', '2019-08-01', '2020-08-01']})
+    external_df.ww.init(name='external')
+
+    es._set_secondary_time_index(external_df, {'dates': ['id']})
+    assert 'external' not in es.dataframe_dict
+    assert external_df.ww.metadata['secondary_time_index'] == {'dates': ['id', 'dates']}
```

Comment:
I'm kind of wondering if we need either of these tests...

The not present error should be covered by other entityset tests for getitem and this isn't something being controlled inside the `set_secondary_time_index` method.

I'm not sure that setting a secondary time index for an external dataframe is a behavior we care about, and since it is through a private method, users shouldn't be doing this type of thing...

Open to discuss though.

Reply:
hmm that's fair. Mostly wrote these tests as a sanity check to make sure the error was clear and to see what happened if we called `_set_secondary_time_index` with an external dataframe, but happy to remove since you're definitely right that they don't add much value

Reply:
removed the tests